agent/consul/discovery_chain_endpoint.go

```diff
@@ -55,11 +57,28 @@ func (c *DiscoveryChain) Get(args *structs.DiscoveryChainRequest, reply *structs
 				return err
 			}
 
+			_, _, config, err := state.CARootsAndConfig(ws)
+			if err != nil {
+				return err
+			} else if config == nil {
+				return errors.New("no cluster ca config setup")
```

Comment:
🤔 Should we change this error to make it more clear to users that they need to enable connect on servers? I think we already have an error string we use when you attempt to fetch root certs with Connect disabled that might be appropriate?

Reply:
I can early-abort the server RPC the with the same error string as ConnectCA stuff:
```
ErrConnectNotEnabled    = errors.New("Connect must be enabled in order to use this endpoint")
```

Reply:
I left the existing error message alone, but practically folks shouldn't end up hitting these specific error messages anymore.